api/client.py

```diff
@@ -57,30 +61,87 @@ def generate(model_name, prompt, system=None, template=None, format="", context=
     except requests.exceptions.RequestException as e:
         print(f"An error occurred: {e}")
         return None, None
+    
+
+# Create a blob file on the server if it doesn't exist.
+def create_blob(digest, file_path):
+    url = f"{BASE_URL}/api/blobs/{digest}"
+
+    # Check if the blob exists
+    response = requests.head(url)
+    if response.status_code != 404:
+        return  # Blob already exists, no need to upload
+
+    # Upload the blob
+    with open(file_path, 'rb') as file_data:
+        requests.post(url, data=file_data)
+
 
 # Create a model from a Modelfile. Use the callback function to override the default handler.
-def create(model_name, model_path, callback=None):
+def create(model_name, filename, callback=None):
     try:
+        file_path = Path(filename).expanduser().resolve()
+        # Read and process the modelfile
+        with open(file_path, 'r') as f:
+            lines = f.readlines()
+
+        processed_lines = []
+        for line in lines:
+            # Skip empty or whitespace-only lines
+            if not line.strip():
+                continue
+            
+            parts = line.strip().split()
+            command, *args = parts
+
+            if command in ["FROM", "ADAPTER"]:
+                path = Path(args[0]).expanduser()
+
+                # Check if path is relative and resolve it
+                if not path.is_absolute():
+                    path = (file_path.parent / path).resolve()
+                
+                # Skip if file does not exist for "model", this is handled by the server
+                if not path.exists() and command == "FROM":
+                    processed_lines.append(line)
+                    continue
+
+                # Calculate SHA-256 hash
+                with open(path, 'rb') as bin_file:
+                    hash = hashlib.sha256()
+                    hash.update(bin_file.read())
+                    digest = f"sha256:{hash.hexdigest()}"
+                
+                # Add the file to the remote server
+                create_blob(digest, path)
+
+                # Replace path with digest in the line
+                line = f"{command} @{digest}\n"
+
+            processed_lines.append(line)
+
+        # Combine processed lines back into a single string
+        modelfile_content = ''.join(processed_lines)
```

Comment:
the above example strips the newlines so join it back with newlines
```suggestion
        modelfile_content = '\n'.join(processed_lines)
```

Reply:
Or instead of joining a list, you can stream the data into an `io.StringIO` or `io.BytesIO`